### PR TITLE
8253473: Javadoc clean up in HttpHandler, HttpPrincipal, HttpContext, and HttpsConfigurator 

### PR DESCRIPTION
**Pull request by pconcannon**
Hi,

Could someone please review my doc-only fix for JDK-8253473: 'Javadoc clean up in HttpHandler, HttpPrincipal, HttpContext, and HttpsConfigurator ' ?

This fix is set of formatting changes intended to clean up the javadoc of the following classes :

`com.sun.net.httpserver.HttpHandler`
`com.sun.net.httpserver.HttpPrincipal`
`com.sun.net.httpserver.HttpContext`
`com.sun.net.httpserver.HttpsConfigurator` 

This issue is a sub-task of [JDK-8252822](https://bugs.openjdk.java.net/browse/JDK-8252822)

Kind regards,
Patrick

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8253473](https://bugs.openjdk.java.net/browse/JDK-8253473): Javadoc clean up in HttpHandler, HttpPrincipal, HttpContext, and HttpsConfigurator 

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/810/head:pull/810`
`$ git checkout pull/810`

### CODE REVIEW

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java [hunk 1]
```diff
@@ -48,10 +50,13 @@ public HttpPrincipal (String username, String realm) {
...
     /**
-     * Compares two HttpPrincipal. Returns <code>true</code>
-     * if <i>another</i> is an instance of HttpPrincipal, and its
-     * username and realm are equal to this object's username
-     * and realm. Returns <code>false</code> otherwise.
+     * Compare two instances of {@code HttpPrincipal}. Returns {@code true} if
+     * <i>another</i> is an instance of {@code HttpPrincipal}, and its username
+     * and realm are equal to this object's username and realm. Returns {@code false}
+     * otherwise.
+     *
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:57]: Isn't this missing an @param for `another` ?

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java [hunk 2]
```diff
@@ -63,41 +68,47 @@ public boolean equals (Object another) {
...
-     * returns the contents of this principal in the form
-     * <i>realm:username</i>
+     * Returns the contents of this principal in the form
+     * <i>realm:username</i>.
+     *
+     * @return the contents of this principal in the form realm:username
      */
     public String getName() {
         return username;
     }
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:78]: Can you log a follow-up bug about this? It seems that the implementation just returns `username`, not `realm:username`.

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java [hunk 3]
```diff
@@ -63,41 +68,47 @@ public boolean equals (Object another) {
...
      */
     public String getRealm() {
         return realm;
     }
 
     /**
-     * returns a hashcode for this HttpPrincipal. This is calculated
-     * as <code>(getUsername()+getRealm().hashCode()</code>
+     * Returns a hashcode for this {@code HttpPrincipal}. This is calculated
+     * as <code>(getUsername()+getRealm().hashCode()</code>.
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:100]: There is a close parenthesis missing after `getRealm()`

**Review by dfuch — CHANGES_REQUESTED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java (hunk 3 above)
Comment (dfuch) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:100]: Also `<code></code>` could be replaced with `{@code }`

**Review by dfuch — CHANGES_REQUESTED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java (hunk 1 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:57]: Well spotted. @param added in commit d8b93d01021d696f9f5272b1e6b34750dc36bcbf

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java (hunk 2 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:79]: I've created an issue for this as requested: https://bugs.openjdk.java.net/browse/JDK-8255584

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java (hunk 3 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java]: Closing parenthesis added and code tag changed in commit: d8b93d01021d696f9f5272b1e6b34750dc36bcbf

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java [hunk 4]
```diff
@@ -55,10 +55,11 @@ public HttpPrincipal(String username, String realm) {
      * and realm are equal to this object's username and realm. Returns {@code false}
      * otherwise.
      *
+     * @param another the object to compare this instance of {@code HttpPrincipal} against
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:58]: And that might require a CSR...

**Review by dfuch — APPROVED** [review]

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java (hunk 4 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:58]: I've created a CSR as requested: https://bugs.openjdk.java.net/browse/JDK-8255594

Comment (dfuch) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpPrincipal.java:58]: CSR Reviewed.

**Pull request closed without merging**